### PR TITLE
'Update on value change' configuration for save attributes rule node

### PR DESCRIPTION
**Pull request by volodymyr-babak**

### CODE REVIEW

File: projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html [hunk 1]
```diff
@@ -19,4 +19,10 @@
     </mat-checkbox>
     <div class="tb-hint" translate>tb.rulenode.send-attributes-updated-notification-hint</div>
   </div>
+  <div>
+    <mat-checkbox formControlName="updateAttributesOnValueChange">
+      {{ 'tb.rulenode.update-attributes-on-value-change' | translate }}
+    </mat-checkbox>
```
Comment (vvlladd28) [inline projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html]: Why a checkbox and not a mat-slider-toggle?

Comment (volodymyr-babak) [reply projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html]: to be in sync with these guys

**Comment by vvlladd28** [comment]
Updated design
image

File: projects/rulenode-core-config/src/lib/locale/rulenode-core-locale.constant.ts [hunk 2]
```diff
@@ -64,6 +64,9 @@ export default function addRuleNodeCoreLocaleEnglish(translate: TranslateService
           'send-attributes-updated-notification-hint': 'Send notification about updated attributes as a separate message to the rule engine queue.',
           'send-attributes-deleted-notification': 'Send attributes deleted notification',
           'send-attributes-deleted-notification-hint': 'Send notification about deleted attributes as a separate message to the rule engine queue.',
+          'update-attributes-on-value-change':'Save attribute only if the value changes',
```
Comment (ashvayka) [inline projects/rulenode-core-config/src/lib/locale/rulenode-core-locale.constant.ts]: update-attributes-only-on-value-change

File: projects/rulenode-core-config/src/lib/components/action/attributes-config.component.ts [hunk 3]
```diff
@@ -40,6 +41,7 @@ export class AttributesConfigComponent extends RuleNodeConfigurationComponent {
       if (value === AttributeScope.CLIENT_SCOPE) {
         this.attributesConfigForm.get('sendAttributesUpdatedNotification').patchValue(false, {emitEvent: false});
       }
+      this.attributesConfigForm.get('updateAttributesOnValueChange').patchValue(false, {emitEvent: false});
```
Comment (ashvayka) [inline projects/rulenode-core-config/src/lib/components/action/attributes-config.component.ts]: updateAttributesOnlyOnValueChange

File: projects/rulenode-core-config/src/lib/components/action/attributes-config.component.ts [hunk 4]
```diff
@@ -30,7 +30,8 @@ export class AttributesConfigComponent extends RuleNodeConfigurationComponent {
     this.attributesConfigForm = this.fb.group({
       scope: [configuration ? configuration.scope : null, [Validators.required]],
       notifyDevice: [configuration ? configuration.notifyDevice : true, []],
-      sendAttributesUpdatedNotification: [configuration ? configuration.sendAttributesUpdatedNotification : false, []]
+      sendAttributesUpdatedNotification: [configuration ? configuration.sendAttributesUpdatedNotification : false, []],
+      updateAttributesOnValueChange: [configuration ? configuration.updateAttributesOnValueChange : false, []]
```
Comment (ashvayka) [inline projects/rulenode-core-config/src/lib/components/action/attributes-config.component.ts]: updateAttributesOnlyOnValueChange

File: projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html [hunk 5]
```diff
@@ -7,16 +7,28 @@
...
-    <mat-checkbox formControlName="sendAttributesUpdatedNotification">
-      {{ 'tb.rulenode.send-attributes-updated-notification' | translate }}
-    </mat-checkbox>
-    <div class="tb-hint" translate>tb.rulenode.send-attributes-updated-notification-hint</div>
+  <div class="tb-form-panel no-padding no-border">
+    <div tb-hint-tooltip-icon="{{ (attributesConfigForm.get('updateAttributesOnValueChange').value
+                                     ? 'tb.rulenode.update-attributes-on-value-change-hint-enabled'
+                                     : 'tb.rulenode.update-attributes-on-value-change-hint') | translate }}"
+         class="tb-form-row same-padding">
+      <mat-slide-toggle class="mat-slide margin" formControlName="updateAttributesOnValueChange">
```
Comment (ashvayka) [inline projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html]: updateAttributesOnlyOnValueChange

File: projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html [hunk 6]
```diff
@@ -7,16 +7,28 @@
...
+        {{ 'tb.rulenode.update-attributes-on-value-change' | translate }}
```
Comment (ashvayka) [inline projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html]: update-attributes-only-on-value-change

File: projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html [hunk 7]
```diff
@@ -7,16 +7,28 @@
...
-    </mat-checkbox>
-    <div class="tb-hint" translate>tb.rulenode.notify-device-hint</div>
-  </div>
-  <div *ngIf="attributesConfigForm.get('scope').value !== attributeScopeMap.CLIENT_SCOPE">
-    <mat-checkbox formControlName="sendAttributesUpdatedNotification">
-      {{ 'tb.rulenode.send-attributes-updated-notification' | translate }}
-    </mat-checkbox>
-    <div class="tb-hint" translate>tb.rulenode.send-attributes-updated-notification-hint</div>
+  <div class="tb-form-panel no-padding no-border">
+    <div tb-hint-tooltip-icon="{{ (attributesConfigForm.get('updateAttributesOnValueChange').value
```
Comment (ashvayka) [inline projects/rulenode-core-config/src/lib/components/action/attributes-config.component.html]: updateAttributesOnlyOnValueChange

**Review by vvlladd28 — APPROVED** [review]

**Pull request merged**